brp-compress

```diff
@@ -11,46 +11,48 @@ cd $RPM_BUILD_ROOT
 COMPRESS="gzip -9 -n"
 COMPRESS_EXT=.gz
 
-for d in ./usr/man/man* ./usr/man/*/man* ./usr/info \
+MANDIRS=$(ls -d ./usr/man/man* ./usr/man/*/man* ./usr/info \
 	./usr/share/man/man* ./usr/share/man/*/man* ./usr/share/info \
 	./usr/kerberos/man ./usr/X11R6/man/man* ./usr/lib/perl5/man/man* \
-	./usr/share/doc/*/man/man* ./usr/lib/*/man/man*
-do
-    [ -d $d ] || continue
-    for f in `find $d -type f`
```

Comment:
You can replace this with one find |read loop like this:

```bash
MANDIR=...

find $MANDIRS -type f  | while read -r  manfile

...

done 
```

Using find with read eliminates both issues  no need to deal with globbing issues in the ls and find statement or the need to call exit.

If no manuals are found the while loop won't run at all and the script exists.